pkg/local_object_storage/metabase/metadata.go

```diff
@@ -167,10 +260,15 @@ func deleteMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID) error {
 	}
 	for i := range ks {
 		if err := metaBkt.Delete(ks[i]); err != nil {
-			return err
+			return 0, err
 		}
 	}
-	return nil
+
+	if !parent.IsZero() {
+		_, _ = deleteMetadata(tx, cnr, parent, true)
```

Comment:
why error is ignored?

Reply:
log is always better than ignoring. But caller already logs child's errors, I'd keep similar records together in the code. Error can be from `bbolt/Bucket.Delete` for both child and parent objects. And we return child's error while ignore parent's ones, although they can be similar. Sounds like unnecessary duality

> err will lead to incorrect counters maintenance

nothing stops us from returning both child size and parent error in this case. The main thing is to process the return correctly

Reply:
> nothing stops us from returning both child size and parent error in this case

golang basis error to me: handling return value when `err != nil`. a child removal was requested, it is done, some side indexes were not removed, warn errors could be found in logs. key goal is achieved: results returned, strange side errors are logged, I do not understand you

Reply:
> a child removal was requested, it is done, some side indexes were not removed, warn errors could be found in logs.

log records appear for both child and parent, but in different places of code. That's what I don't see the point in. As well as in returning an error for child and ignoring for parent. They can behave the same way, either with returned error or internal log